### PR TITLE
fix: NGSI-LD v1.2.1 and v1.3.1 success response codes on entities creation / update

### PR DESCRIPTION
**Pull request by bobeal**
- In v1.2.1, response code is 200 (the current official version)
- In v1.3.1, response code is 201 if some created entities, 204 if just updated existing (the soon to be released version)

This PR aims at handling these differences in the response code when creating or updating entities in the context broker.

### CODE REVIEW

File: lib/services/devices/devices-NGSI-LD.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ function createInitialEntityHandlerNgsiLD(deviceData, newDevice, callback) {
             alarms.raise(constants.ORION_ALARM, error);
 
             callback(error);
-        } else if (response && response.statusCode === 204) {
+        } else if (response && (response.statusCode === 200 || response.statusCode == 201 || response.statusCode == 204)) {
```
Comment (fgalan) [inline lib/services/devices/devices-NGSI-LD.js:79]: Maybe a comment above this line should explain which spec version uses each "flavour".

File: lib/services/devices/devices-NGSI-LD.js [hunk 2]
```diff
@@ -118,7 +118,7 @@ function updateEntityHandlerNgsiLD(deviceData, updatedDevice, callback) {
             alarms.raise(constants.ORION_ALARM, error);
 
             callback(error);
-        } else if (response && response.statusCode === 204) {
+        } else if (response && (response.statusCode === 200 || response.statusCode == 204)) {
```
Comment (fgalan) [inline lib/services/devices/devices-NGSI-LD.js:121]: Maybe a comment above this line should explain which spec version uses each "flavour".

**Comment by fgalan** [comment]
Maybe the new cases introduced by this PR should be covered by unit tests.

**Comment by fgalan** [comment]
(Closed by error, sorry. Re-opened).

**Comment by fgalan** [comment]
LGTM from @jason-fox would be derivable in this PR.

File: lib/services/devices/devices-NGSI-LD.js (hunk 1 above)
Comment (bobeal) [reply lib/services/devices/devices-NGSI-LD.js:79]: I tend not to overload the code when the info is easily findable in Git (I've put all the details in the commit message). But as you wish 😄

**Comment by bobeal** [comment]
> Maybe the new cases introduced by this PR should be covered by unit tests.

Yes, why not.

I was wondering where to put them. In a new test file in the `test/unit/ngsi-ld/general` directory?

**Comment by jason-fox** [comment]
The new test cases should be added to:

- `/test/unit/ngsi-ld/provisioning/device-provisioning-api_test.js`
- `/test/unit/ngsi-ld/ngsiService/active-devices-test.js`

**Comment by jason-fox** [comment]
LGTM

**Comment by jason-fox** [comment]
@fgalan Please merge

File: lib/services/devices/devices-NGSI-LD.js (hunk 1 above)
Comment (fgalan) [reply lib/services/devices/devices-NGSI-LD.js:79]: Not everybody is used to check commit messages to look for information ;), so I think a similar comment in the code doesn't hurt and could help, pls.

File: lib/services/devices/devices-NGSI-LD.js (hunk 2 above)
Comment (fgalan) [reply lib/services/devices/devices-NGSI-LD.js:121]: Same here...

**Comment by fgalan** [comment]
> @fgalan Please merge

I have realized of a couple of comment threads yet opened. I think they involve minor modification. @bobeal what do you think?

**Comment by bobeal** [comment]
> I have realized of a couple of comment threads yet opened. I think they involve minor modification. @bobeal what do you think?

Indeed, I've forgotten to deal with them. Doing it right now.

File: lib/services/devices/devices-NGSI-LD.js (hunk 1 above)
Comment (bobeal) [reply lib/services/devices/devices-NGSI-LD.js]: Added some comments.

**Comment by fgalan** [comment]
> Indeed, I've forgotten to deal with them. Doing it right now.

Nice comment! :)

**Pull request merged**